core/src/UVMap.cpp

```diff
@@ -138,6 +149,119 @@ auto UVMap::Plot(
     return r;
 }
 
+// pearson correlation coefficient between two vectors
+static auto pearson_correlation(
+    const std::vector<double>& x, const std::vector<double>& y) -> double
+{
+    auto n = x.size();
+    auto sum_x = std::accumulate(x.begin(), x.end(), 0.0);
+    auto sum_y = std::accumulate(y.begin(), y.end(), 0.0);
+    auto sum_xy = std::inner_product(x.begin(), x.end(), y.begin(), 0.0);
+    auto sum_x2 = std::inner_product(x.begin(), x.end(), x.begin(), 0.0);
+    auto sum_y2 = std::inner_product(y.begin(), y.end(), y.begin(), 0.0);
+
+    auto numerator = n * sum_xy - sum_x * sum_y;
+    auto denominator =
+        std::sqrt((n * sum_x2 - sum_x * sum_x) * (n * sum_y2 - sum_y * sum_y));
+
+    auto p = numerator / denominator;
+    if (std::isnan(p)) {
+        throw std::runtime_error("pearson correlation coefficient is NaN");
+    } else {
+        return p;
+    }
+}
```

Comment:
Wrap this in an anonymous namespace instead of using `static`:
```c++
// definition
namespace {
    auto PearsonCorrelation() { ... } -> double;
}

// usage
::PearsonCorrelation();
```